main/acle.rst

```diff
@@ -5822,3 +5835,183 @@ Instructions
 +---------------------------------+----------------+------------+-----------------+
 
 These intrinsics are available when ``arm_acle.h`` is included.
+
+Architectural Extension Bridges
+###############################
+
+.. _ssec-NEONSVEBridge:
+
+NEON-SVE Bridge
+===============
+
+
+The NEON_SVE Bridge adds intrinsics that allow conversions between NEON and
+SVE vectors.
+
+These intrinsics are defined in the header file ``arm_neon_sve_bridge.h``.
+
+Including this header_file will include ``arm_neon.h`` and ``arm_sve.h``.
+
+Including this header file will define ``__ARM_NEON_SVE_BRIDGE`` as 1.
```

Comment:
Yep - good point @rsandifo-arm ! I think that we should raise an error if the header file is included but the compiler is not defining the `__ARM_NEON_SVE_BRIDGE`. The macro should be defined every time SVE is present, therefore it seems to me that it should have the same meaning of `__ARM_FEATURE_SVE`. 

This might mean that we don't need to use any feature macro, and make the bridge available unconditionally every time we include the header file and we target SVE or SVE2 with the compiler?

Given that the bridge will include `arm_sve.h`, the error will be pretty clear if we want to use the bridge in a compiler invocation that is not targeting SVE.

Reply:
I guess the question is how best to allow the user to know whether `#include "arm_neon_sve_bridge.h"` will work or not, given that compilers with `arm_sve.h` and without the bridge already exist. If `__has_include` or similar is present then I agree it is possibly redundant, but I don't know if there is a precedent for relying on that?

Reply:
Yeah, AIUI, the feature macros are provided for the reason that @georges-arm mentioned.  E.g. the canonical way of including `arm_sve.h` is:
```
#ifdef __ARM_FEATURE_SVE
#include <arm_sve.h>
#endif
```
I'm pretty sure no-one does that, but it's the canonical way nonetheless.

The idea would be the same here.  The compiler defines `__ARM_NEON_SVE_BRIDGE` to advertise to the source code that the compiler provides an `arm_neon_sve_bridge.h`.  The code can then include the header file when available and (if it wants to) fall back to something else otherwise.

Reply:
I see, OK. So we do the following:

1. we introduce the macro in the specs but not in the header file;
2. we require the compiler to set it to 1 when targeting SVE;
3. We raise an error if the macro is not active and the header file is included via `#ifndef __ARM_NEON_SVE_BRIDGE #error "..." #endif` in the header file.

Reply:
1. 👍 
2. We're probably in agreement and I'm probably just being overly pedantic, but: compilers should only set the macro to 1 if they provide the new header file.  Old compilers that target SVE would still be conforming (they would not define the macro and would not provide the header file).
3. I don't think the error is necessary.  The macro is only there to tell the user whether a feature is available.

Reply:
done.